python/GafferImageUI/CatalogueUI.py

```diff
@@ -880,4 +910,58 @@ def __keyPress( self, imageListing, keyEvent ) :
 
 		return False
 
+	def __columnContextMenuDefinition( self ) :
+
+		weakSelf = weakref.ref( self )
+
+		def toggleColumn( column, state ) :
+
+			if weakSelf() is None :
+				return
+
+			columns = list( weakSelf().__getColumns() )
+			if state :
+				columns.append( column )
+			else :
+				columns.remove( column )
+
+			weakSelf().__setColumns( columns )
+
+		columns = self.__getColumns()
+		allColumnsSorted = sorted( registeredColumns() )
+
+		menu = IECore.MenuDefinition()
+
+		menu.append( "/Reset", { "command" : Gaffer.WeakMethod( self.__resetColumns ) } )
+		menu.append( "/__resetDivider__", { "divider" : True } )
+
+		for column in allColumnsSorted :
+
+			definition = columnDefinition( column )
+			path = "/".join( [ IECore.CamelCase.toSpaced(p) for p in column.split( "/") ])
```

Comment:
Let's use the name verbatim, and register with the exact user-visible names we want in the first place. That's what we do for presets and outputs and so on...

Reply:
Fixed https://github.com/GafferHQ/gaffer/pull/3688/commits/78ba11e3bde4ea782b13c5a708537b676d05cb8b#diff-f8b21d2701a672983fe238345811269bR979, and in all other column registrations.